examples/experimental/horizontal-pod-autoscaling/README.md

```diff
@@ -0,0 +1,189 @@
+This can be simply achieved on GCE by exporting ```KUBE_ENABLE_HORIZONTAL_POD_AUTOSCALER=true``` before running ```kube-up.sh``` script.
+
+The required version of kubectl is also 1.1.
+
+
+## Step One: Run & expose php-apache server
+
+To demonstrate horizontal pod autoscaler we will use a custom docker image based on php-apache server.
+The image can be found [here](image/).
+It defines [index.php](image/index.php) page which performs some CPU intensive computations.
+
+First, we will start a replication controller running the image and expose it as an external service:
+
+```console
+$ kubectl run php-apache --image=gcr.io/dev-jsz/php-apache --replicas=1
+replicationcontroller "php-apache" created
+
+$ kubectl expose rc php-apache --port=80 --type=LoadBalancer
+
+```
+
+Now, we will wait some time and verify that both the replication controller and the service were correctly created and are running. We will also determine the IP address of the service:
+
+```console
+$ kubectl get pods
+NAME               READY     STATUS    RESTARTS   AGE
+php-apache-wa3t1   1/1       Running   0          12m
+
+$ kubectl describe services php-apache | grep "LoadBalancer Ingress"
+LoadBalancer Ingress:	146.148.24.244
+
+```
+
+We may now check that php-apache server works correctly by calling ```curl``` with the service's IP:
+
+```console
+$ curl http://146.148.24.244
+OK
+```
+
+## Step Two: Create horizontal pod autoscaler
+
+Now that the server is running, we will create a horizontal pod autoscaler for it.
+To create it, we will use [hpa-php-apache.yaml](hpa-php-apache.yaml) file, which look like this:
+
+```yaml
+apiVersion: experimental/v1alpha1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: php-apache
+  namespace: default
+spec:
+  maxReplicas: 10
+  minReplicas: 1
+  scaleRef:
+    kind: ReplicationController
+    name: php-apache
+    namespace: default
+  target:
+    quantity: 100m
+    resource: cpu
+```
+
+Please note that it sets the target CPU consumption to 100 mili-cores.
+
+We will create the autoscaller by executing the following command:
+
+```console
+$ kubectl create -f ./examples/experimental/horizontal-pod-autoscaling/hpa-php-apache.yaml
+horizontalpodautoscaler "php-apache" created
+```
+
+We may check the current status of autoscaler by running:
+
+```console
+$ kubectl get hpa
+NAME         REFERENCE                                   TARGET     CURRENT   MINPODS   MAXPODS   AGE
+php-apache   ReplicationController/default/php-apache/   100m cpu   0 cpu     1         10        4m
+```
+
+Please not that the current CPU consumption is 0 as we are not sending any requests to the server.
+
+## Step Three: Increase load
+
+Now, we will see how the autoscaler reacts on the increased load of the server.
+We will start an infinite loop of queries to our server (please run it in a different terminal):
+
+```console
+$ while true; do curl http://146.148.6.244; done
+```
+
+We may examine, how CPU load was increased by executing:
```

Comment:
Comment on the latency of the measurement?  Maybe just say do a ```watch kubectl get hpa``` and you will see the observed load increase?